libraries/AP_Devo_Telem/AP_Devo_Telem.h

```diff
@@ -0,0 +1,80 @@
+// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __AP_DEVO_TELEM_H__
+#define __AP_DEVO_TELEM_H__
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_Baro/AP_Baro.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+
+
+#define DEVOM_SYNC_BYTE        0xAA
+
+#pragma pack(push, 1)
+typedef struct {
+        uint8_t header;                                ///< 0xAA for a valid packet
+        int32_t lon;
+        int32_t lat;
+        int32_t alt;
+        int16_t speed;
+        int16_t temp;
+        int16_t volt;
+        uint8_t checksum8;   ///< CRC8
+} DevoMPacket;
+#pragma pack(pop)
+
+class AP_DEVO_Telem
+{
+public:
+    static AP_DEVO_Telem create(AP_AHRS &ahrs, const AP_BattMonitor &battery) {
+        return AP_DEVO_Telem{ahrs, battery};
+    }
```

Comment:
We have removed the create methods, so remove this. You should add the deletion of copy constructors though:
https://github.com/ArduPilot/ardupilot/blob/ee8adf90730002b4211bd36dd289d60d8fe4c0a5/libraries/AP_GPS/AP_GPS.h#L69-L71

Reply:
Can you please add that code? You forgot to copy that from FrSky...

Reply:
Ok, but there was no such code in FrSky when I did this library (3 years or so)

Reply:
Yeah, we aren't stopping development, things evolve.

Reply:
But allow PR to hang for years

Reply:
Yes, when the OP doesn't comply to our rules and let's it hang for years. We could just close it but we prefer to believe people are smart enough to make the asked changes.

Reply:
no comments, no reviews, nothing. just a spherical PR in vacuum

Reply:
No need to tell the same person the same thing over and over. OP is given the contribution rules, is told what's wrong with it, if OP opens another PR doing the same, then that's a dumb move - and yes, it won't get anyone looking at it, it's not worth it.

Reply:
really? seems that you try to отмазаться

Reply:
this pr - #3672 - opened at the begining of 2016 (not my, just as example). No changes requestes, just hangs. PR is really useful, so why?

Reply:
> really? seems that you try to отмазаться

Please write in English, otherwise I might think you're insulting me and I'll have to block you.

> this pr - #3672 - opened at the begining of 2016 (not my, just as example). No changes requestes, just hangs. PR is really useful, so why?

Really? There is a long discussion there plus more in some open issues... Keep trying.

Reply:
> I might think you're insulting me and I'll have to block you

you can use Google Chrome with automatic translation to avoid wrong thinking

Reply:
>There is a long discussion there plus more in some open issues

I tested this PR in my fork and it works just fine. Did you test it in real flight?

Reply:
> you can use Google Chrome with automatic translation to avoid wrong thinking

No, Google Chrome won't offer translation here. And Google Translate can't translate that word. So...

> I tested this PR in my fork and it works just fine. Did you test it in real flight? It is really good in early diagnosis of ESC/motor issues.

ArduPilot is working just fine for thousands of people, does it means it's bug free? No.
I'm not going to discuss the merits of that PR, it was discussed and vehicle maintainer wasn't or isn't sure it is worth it so it hasn't been merged yet.
If we acted like you want all your PRs would have been dismissed as you didn't follow our basic contribution rules nor wanted to when requested.

Reply:
>Google Translate can't translate that word

it means 'to justify one's own position by unsuitable arguments'. Nothing insulting.

> you didn't follow our basic contribution rules

I try to follow, but not all changes requested are caused by that rules, isn't it? Eg. to replace GPS coordinates by AHRS ones - no rules requires it.